### ISSUE TITLE
glz::read format glz::csv infinite loop and memory exhaustion

### ISSUE THREAD

**Issue by arturbac** [stephenberry/glaze#768]
reading csv with invalid file format is causing infinite loop and when last column is string also memory exhaustion without issues reported from asan or ubsan. Expected by me would be getting syntax error with invalid files.

```
struct glaze_test_struct_t
  {
  std::vector<int> num1{};
  std::vector<std::string> str1{};
  };
constexpr std::string_view invalid_record_1 =
  R"(num1,str1
11,Warszawa
33,Krakow,some text,
55,Gdynia
77,Reda)";

constexpr std::string_view invalid_record_2 =
  R"(num1,str1
11,Warszawa
33,Krakow,some text
55,Gdynia
77,Reda)";
static void test_galze_internal()
  {

  "glz invalid_record_1"_test = []
  {
    glz::context ctx{};
    glaze_test_struct_t value;
    glz::parse_error glaze_err{
      glz::read<glz::opts{.format = glz::csv, .layout = glz::colwise}>(value, std::string{invalid_record_1}, ctx)
    };
    ut::expect(bool(glaze_err));
  };

  "glz invalid_record_2"_test = []
  {
    glz::context ctx{};
    glaze_test_struct_t value;
    glz::parse_error glaze_err{
      glz::read<glz::opts{.format = glz::csv, .layout = glz::colwise}>(value, std::string{invalid_record_2}, ctx)
    };
    ut::expect(bool(glaze_err));
  };
  }
```

**Comment by stephenberry** [comment]
Thanks for pointing out this bug. Will work on fixing it.

**Comment by arturbac** [comment]
The problem is  in logic here
Instead of verification and expecting end of line You are testing for end of line and if that not happens row is not incremented and another set of keys is read from completely wrong position
for csv those conditionals maybe should be changed to expectations and report failure ..
![obraz](https://github.com/stephenberry/glaze/assets/14975842/5c17964c-5128-4fb8-9e24-f9799fcabfe7)

**Comment by arturbac** [comment]
Yep this fixes code and tests for me maybe not efficient but correct
![obraz](https://github.com/stephenberry/glaze/assets/14975842/80f7109a-7f3d-4541-a217-11292f043fc3)

**Comment by arturbac** [comment]
This would be more efficient.
![obraz](https://github.com/stephenberry/glaze/assets/14975842/5d7b6ee7-943d-4bb0-8cf8-4a4b745b27fc)

**Comment by arturbac** [comment]
PR or You want to fix it Yourself ?

**Comment by stephenberry** [comment]
It will probably be fixed faster if you're able to issue a PR

**Comment by arturbac** [comment]
ok, I will work on it.

**Comment by stephenberry** [comment]
Fixed in #771
